src-theme/src/theme/arraylist.ts

```diff
@@ -0,0 +1,31 @@
+import {getModuleSettings} from "../integration/rest";
+import type {ChoiceSetting, ChooseSetting, ConfigurableSetting,} from "../integration/types";
+
+export async function getPrefixAsync(name: string) {
```

Comment:
This method suffers multiple problems:
1. Use if-guard-patterns
2. Split this function into two functions: `getPrefixName` which returns the prefix without leading space or null and `formatPrefixName` which formats this into the final value

But I think there is a better way to implement it: Add an option to configurables which allows the mod to specify which setting should be displayed as prefix. Go through the configurables recursively and return the first configurable which has it's prefix flag set to true.

Fyi, if-guard-patterns are:
```JavaScript
let choice = ...;

if (mode != null) {
    return Promise.resolve(cMode.active);
}

let configurable = ...;

if (configurable == null) {
    return Promise.resolve(null);
}

...
```

Reply:
> This method suffers multiple problems:
> 
>     1. Use if-guard-patterns
> 
>     2. Split this function into two functions: `getPrefixName` which returns the prefix without leading space or null and `formatPrefixName` which formats this into the final value
> 
> 
> But I think there is a better way to implement it: Add an option to configurables which allows the mod to specify which setting should be displayed as prefix. Go through the configurables recursively and return the first configurable which has it's prefix flag set to true.
> 
> Fyi, if-guard-patterns are:
> 
> ```js
> let choice = ...;
> 
> if (mode != null) {
>     return Promise.resolve(cMode.active);
> }
> 
> let configurable = ...;
> 
> if (configurable == null) {
>     return Promise.resolve(null);
> }
> 
> ...
> ```

i did what you suggested, but since i know absolutely nothing about kotlin im keeping the shitty method for now